packages/wrangler/src/git-client.ts

```diff
@@ -0,0 +1,43 @@
+import { execa } from "execa";
+import { findUp } from "find-up";
+
+/**
+ * Check whether the given current working directory is within a git repository
+ * by looking for a `.git` directory in this or an ancestor directory.
+ */
+export async function isInsideGitRepo(cwd: string) {
+  const res = await findUp(".git", { cwd, type: "directory" });
+  return res !== undefined;
+}
+
+/**
+ * Check whether git is installed by trying to run it.
+ */
+export async function isGitInstalled() {
+  try {
+    return (await execa("git", ["--version"])).exitCode === 0;
+  } catch (err) {
+    return false;
+  }
+}
+
+/**
+ * Initialize a new Worker project with a git repository.
+ *
+ * We want the branch to be called `main` but to support earlier versions of git we cannot use
+ * `git init --initial-branch`.
+ * Instead, we rename the branch to `main` using `git branch -m`.
+ * But this can fail if there are no commits on the current branch so we create an artificial
+ * empty commit to guard against that.
+ */
+export async function initializeGit(cwd: string) {
+  await execa("git", ["init"], {
```

Comment:
are we not using `git init -b main` for better backward compatibility?

Reply:
`init -b` doesn't work on older git versions 😢  see https://github.com/cloudflare/wrangler2/issues/1168